blocks/bricks/__init__.py

```diff
@@ -158,7 +159,56 @@ def _push_initialization_config(self):
                     child.biases_init = self.biases_init
 
 
-class Linear(Initializable):
+class Feedforward(Brick):
+    """Declares an interface for bricks with one input and one output.
+
+    Many bricks have just one input and just one output (activations,
+    :class:`Linear`, :class:`MLP`). To make such bricks interchangable
+    in most contexts they should share an interface for configuring
+    their input and output dimensions. This brick provides such an
+    interface.
+
+    """
+    @abstractproperty
+    def input_dim():
+        """The input dimension of the brick."""
+        pass
+
+    @abstractproperty
+    def output_dim():
+        """The output dimension of the brick."""
+        pass
+
+
+def override_with_attribute(property_, attribute=None):
```

Comment:
If you just want it to given an error on access, then I don't see why the usual way doesn't work:
```python
@property
def input_dim(self):
    if not hasattr(self, '_input_dim'):
        raise ValueError("Your feed-forward brick forgot to set the input_dim")
    return self._input_dim

@input_dim.setter
def input_dim(self, value):
    self._input_dim = value
```
Either way, the user is allowed to change `input_dim` or `output_dim` up to allocation, right? So I think that this is definitely the best option. An `abstractproperty` doesn't make sense if you are allowing people to set attributes instead of properties.